src/Interpreters/Aggregator.cpp

```diff
@@ -236,15 +236,19 @@ void initDataVariantsWithSizeHint(
             else
             {
                 const auto adjusted = std::max(lower_limit, hint->median_size);
```

Comment:
This can also be inside the new if block and you could make `else`  ->  `else if`.
Also I see in the above if block we log a message if we dont preallocate space, would it be helpful to also log if we dont preallocate because of max_threads & size too ? what do you think?

Reply:
> what do you think?

I find this a bit excessive. it is not really user visible behaviour and it is not actionable (user cannot change this behaviour using some setting).